src/o365/spo/commands/folder/folder-move.ts

```diff
@@ -0,0 +1,331 @@
+          }),
+          body: {
+            exportObjectUris: [sourceAbsoluteUrl],
+            destinationUri: this.urlCombine(tenantUrl, args.options.targetUrl),
+            options: {
+              "AllowSchemaMismatch": allowSchemaMismatch,
+              "IgnoreVersionHistory": true,
+              "IsMoveMode": true,
+            }
+          },
+          json: true
+        };
+
+        if (this.debug) {
+          cmd.log('CreateCopyJobs request...');
+          cmd.log(requestOptions);
+        }
+
+        return request.post(requestOptions);
+      })
+      .then((jobInfo: any): Promise<any> => {
+        if (this.debug) {
+          cmd.log('CreateCopyJobs response...');
+          cmd.log(jobInfo);
+        }
+
+        const jobProgressOptions: JobProgressOptions = {
+          webUrl: webUrl,
+          accessToken: siteAccessToken,
+          copyJopInfo: jobInfo.value[0],
+          progressMaxPollAttempts: 1000, // 1 sec.
+          progressPollInterval: 30 * 60, // approx. 30 mins. if interval is 1000
+          progressRetryAttempts: 5
+        }
+
+        return this.getCopyJobProgress(jobProgressOptions, cmd);
+      })
+      .then((): void => {
+        if (this.verbose) {
+          cmd.log('DONE');
+        }
+
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  /**
+   * A polling function that awaits the 
+   * Azure queued copy job to return JobStatus = 0 meaning it is done with the task.
+   */
+  private getCopyJobProgress(opts: JobProgressOptions, cmd: CommandInstance):
+    Promise<void> {
+
+    let pollCount: number = 0;
+    let retryAttemptsCount: number = 0;
+
+    const checkCondition = (resolve: () => void, reject: (error: any) => void): void => {
+      pollCount++;
+      const requestUrl: string = `${opts.webUrl}/_api/site/GetCopyJobProgress`;
+      const requestOptions: any = {
+        url: requestUrl,
+        headers: Utils.getRequestHeaders({
+          authorization: `Bearer ${opts.accessToken}`,
+          'accept': 'application/json;odata=nometadata'
+        }),
+        body: { "copyJobInfo": opts.copyJopInfo },
+        json: true
+      };
+
+      if (this.debug) {
+        cmd.log('getCopyJobProgress request...');
+        cmd.log(requestOptions);
+      }
+
+      request.post(requestOptions).then((resp: any): void => {
+        retryAttemptsCount = 0; // clear retry on promise success 
+
+        if (this.debug) {
+          cmd.log('getCopyJobProgress response...');
+          cmd.log(resp);
+        }
+
+        if (this.verbose) {
+          if (resp.JobState && resp.JobState === 4) {
+            cmd.log(`Check #${pollCount}. Copy job in progress... JobState: ${resp.JobState}`);
+          }
+          else {
+            cmd.log(`Check #${pollCount}. JobState: ${resp.JobState}`);
+          }
+        }
+
+        for (const item of resp.Logs) {
+          const log = JSON.parse(item);
```

Comment:
We should avoid using `any` and use defined types instead. This will make the code clearer, more robust and less error-prone in refactorings.

Reply:
I confess, the 'any's might be left by me there long time ago in the file copy command (since that command could be a clone of the copy command), but we should definately change that if possible.